### PR TITLE
Fix hovers in the HTML language server

### PR DESCRIPTION
**Pull request by chrismwendt**
Hovers were broken (e.g. on https://github.com/daxeel/blockshell/blob/HEAD/templates/blocks.html):

image

This was happening because https://github.com/vscode-langservers/vscode-html-languageserver was assuming that the initialize params were either null or had an `embeddedLanguages` field, but Sourcegraph passes an object without the `embeddedLanguages` field. This was causing an undefined value to be passed around and eventually it caused this error.

I forked https://github.com/chrismwendt/vscode-html-languageserver and https://github.com/chrismwendt/vscode-html-languageserver-bin in order to fix this. See the comparison pages for details:

- https://github.com/vscode-langservers/vscode-html-languageserver/compare/master...chrismwendt:master
- https://github.com/vscode-langservers/vscode-html-languageserver-bin/compare/master...chrismwendt:master

### CODE REVIEW

**Comment by felixfbecker** [comment]
We should avoid depending on forks. Other langservers might have problems woth initializeOptions too. I believe the current way we use it (passing config) should go through didChangeConfiguration. Is there some other way we can fix this on our end to avoid forking whenever a langserver uses initializeOptions?

**Comment by chrismwendt** [comment]
I submitted https://github.com/vscode-langservers/vscode-html-languageserver/pull/1

This problem is specific to this particular language server and I don't see how we can prevent this problem in a general way.

File: dockerfiles/html/Dockerfile [hunk 1]
```diff
@@ -18,7 +15,16 @@ USER node
 # see https://github.com/nodejs/docker-node/blob/master/docs/BestPractices.md#global-npm-dependencies
 ENV NPM_CONFIG_PREFIX=/home/node/.npm-global
 ENV PATH=$PATH:/home/node/.npm-global/bin
-RUN npm i -g vscode-html-languageserver-bin@1.3.0
+# Ignore the warning about using WORKDIR instead of cd for convenience.
+# hadolint ignore=DL3003
+RUN cd "$HOME" \
+  && git clone https://github.com/chrismwendt/vscode-html-languageserver-bin --recursive \
```
Comment (felixfbecker) [inline dockerfiles/html/Dockerfile]: Can you shallow-clone only the commit?

Comment (chrismwendt) [reply dockerfiles/html/Dockerfile]: Yes, but that'll probably only save a few megs and won't be much faster.

**Pull request merged**